modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/criteria/contributor/UserSegmentsCriteriaContributor.java

```diff
@@ -65,21 +65,28 @@ public void contribute(
 
 		String newFilterString = filterString;
 
-		Matcher matcher = _pattern.matcher(filterString);
+		Matcher matcher1 = _pattern1.matcher(filterString);
 
-		while (matcher.find()) {
-			long roleId = _getRoleId(matcher.group());
+		while (matcher1.find()) {
+			long roleId = _getRoleId(matcher1.group());
 
 			newFilterString = StringUtil.replace(
-				newFilterString, matcher.group(),
+				newFilterString, matcher1.group(),
 				String.join(
 					StringPool.BLANK, StringPool.OPEN_PARENTHESIS,
-					matcher.group(), _getGroupIdsFilterString(roleId),
+					matcher1.group(), _getGroupIdsFilterString(roleId),
 					_getOrganizationIdsFilterString(roleId),
 					_getUserGroupIdsFilterString(roleId),
 					StringPool.CLOSE_PARENTHESIS));
 		}
 
+		Matcher matcher2 = _pattern2.matcher(filterString);
```

Comment:
If you are not using `matcher1` after this line, consider rename it back to `matcher` and set its value to `_pattern2.matcher(filterString)`.